app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/provision/options_helper.rb

```diff
@@ -0,0 +1,13 @@
+module ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::Provision::OptionsHelper
+  def cloud_instance_id
+    source.ext_management_system.uid_ems
+  end
+
+  def vm_image
+    @vm_image ||= MiqTemplate.find_by(:id => get_option(:src_vm_id))
+  end
+
+  def sap_image?
+    vm_image.description == 'stock-sap'
```

Comment:
Would it be better to check the vm_image flavor class name instead of using the description?

We could even add a method to your template subclass:
```
class ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::Template
  def sap_image?
    flavor.kind_of?(ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::SAPProfile)
  end
```

Reply:
I'm not sure I understand fully. `vm_image` here doesn't have a direct flavor association here right (it's just a Template/image)? The user selected flavor is included in the clone_options, and could be queried directly in the cloning module.

Reply:
Ah yes you're right, sorry idk why I thought your images had flavors

Reply:
Past the details here, though, I like the idea of checking a type rather than comparing string.